Userland/Libraries/LibCore/Directory.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2022, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "Directory.h"
+#include "DirIterator.h"
+#include "System.h"
+#include <dirent.h>
+
+namespace Core {
+
+// We assume that the fd is a valid directory.
+Directory::Directory(int fd, Optional<LexicalPath> path)
+    : m_path(move(path))
+    , m_directory_fd(fd)
+{
+}
+
+static ErrorOr<bool> is_valid_directory(int fd)
```

Comment:
there's a same function `Core::File`. i guess you have to remove it

https://github.com/SerenityOS/serenity/blob/751b605690fa28540243a347836c19b972362f20/Userland/Libraries/LibCore/File.cpp#L118-L124

Reply:
Separate commit but sure